src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/ApplicationController.cs

```diff
@@ -299,12 +305,6 @@ public async Task<IActionResult> Edit(EditOpenIdApplicationViewModel model, stri
                     return NotFound();
                 }
 
-                if (!model.UpdateClientSecret && await _applicationManager.IsPublicAsync(application) &&
```

Comment:
Any reason to remove this check? If we don't have it, this means you'll get an exception for apps that are currently public but are updated to be confidential if you don't specify a new secret.

Reply:
I'll look into it

Reply:
I've tested it, and I do not get an exception.
The check is moved up to line 280, so we removed this redundant check.